### ISSUE TITLE
Getting "panic: runtime error: invalid memory address or nil pointer dereference" on 0.7.0

### ISSUE THREAD

**Issue by sandbergja** [daveshanley/vacuum#412]
This started failing for me on 0.7.0.  Previous versions of vacuum did not give the same error.

Steps to recreate:

1. `wget https://raw.githubusercontent.com/pulibrary/allsearch_api/main/swagger/v1/swagger.yaml`
2. `npm i -g @quobix/vacuum@0.7.0`
3. `npx vacuum lint swagger.yaml` <-- gives an error on 0.7.0
4. `npm uninstall -g @quobix/vacuum@0.7.0 && npm i -g @quobix/vacuum@0.6.3`
5. `npx vacuum lint swagger.yaml` <-- gives no error on 0.6.3

Here is the full error message:

```
 INFO  Linting file 'swagger.yaml' against 43 rules: https://quobix.com/vacuum/rulesets/recommended

panic: panic: runtime error: invalid memory address or nil pointer dereference
stacktrace:
goroutine 1687 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.21.5/x64/src/runtime/debug/stack.go:24 +0x64
github.com/sourcegraph/conc/panics.NewRecovered(0x0?, {0x103147b00, 0x103ebc420})
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/panics/panics.go:59 +0x70
github.com/sourcegraph/conc/panics.(*Catcher).tryRecover(0x140006c6328)
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/panics/panics.go:28 +0x58
panic({0x103147b00?, 0x103ebc420?})
	/opt/hostedtoolcache/go/1.21.5/x64/src/runtime/panic.go:914 +0x218
github.com/daveshanley/vacuum/functions/openapi.ExamplesSchema.RunRule.func8()
	/home/runner/work/vacuum/vacuum/functions/openapi/examples_schema.go:200 +0x64
github.com/sourcegraph/conc/panics.(*Catcher).Try(0x0?, 0x0?)
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/panics/panics.go:23 +0x50
github.com/sourcegraph/conc.(*WaitGroup).Go.func1()
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/waitgroup.go:32 +0x58
created by github.com/sourcegraph/conc.(*WaitGroup).Go in goroutine 1555
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/waitgroup.go:30 +0x7c



goroutine 1555 [running]:
github.com/sourcegraph/conc/panics.(*Catcher).Repanic(...)
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/panics/panics.go:38
github.com/sourcegraph/conc.(*WaitGroup).Wait(0x140006c6318)
	/home/runner/go/pkg/mod/github.com/sourcegraph/conc@v0.3.0/waitgroup.go:42 +0x5c
github.com/daveshanley/vacuum/functions/openapi.ExamplesSchema.RunRule({}, {0x30?, 0x1031c28c0?, 0x64000140004caa98?}, {0x14000827aa0, 0x14000091c80, {0x1030c9d00, 0x1033177e0}, {0x0, 0x0}, ...})
	/home/runner/work/vacuum/vacuum/functions/openapi/examples_schema.go:206 +0x414
github.com/daveshanley/vacuum/motor.buildResults({0x14000091c80, 0x1400042f900, 0x1400042f900, {0x103322f48, 0x1400049a000}, 0x14000127a28, 0x14000676198, 0x140002c0500, 0x1400015ab00, 0x0, ...}, ...)
	/home/runner/work/vacuum/vacuum/motor/rule_applicator.go:770 +0x588
github.com/daveshanley/vacuum/motor.runRule({0x14000091c80, 0x1400042f900, 0x1400042f900, {0x103322f48, 0x1400049a000}, 0x14000127a28, 0x14000676198, 0x140002c0500, 0x1400015ab00, 0x0, ...}, ...)
	/home/runner/work/vacuum/vacuum/motor/rule_applicator.go:692 +0x5dc
created by github.com/daveshanley/vacuum/motor.ApplyRulesToRuleSet.func7 in goroutine 1516
	/home/runner/work/vacuum/vacuum/motor/rule_applicator.go:553 +0x21c

```

Thanks for looking into this!  Let me know if there's anything I can do to help.

**Comment by daveshanley** [comment]
Hi, 0.7.0 has significant changes in it, for multiple reasons, this is likely an uncaught regression. 

I can help diagnose this, but in order to do so, I will need the specification you're using, so I can trace the problem. 

If you don't want to share it in public, please email it to me :)

**Comment by daveshanley** [comment]
This has been resolved in `v0.7.1`

![Screenshot 2024-01-12 at 9 58 19 AM](https://github.com/daveshanley/vacuum/assets/187345/594a538c-4155-4d3b-a9e8-2422aeba27a7)

**Comment by sandbergja** [comment]
Works like a charm, thanks @daveshanley !
